inc/functions/api.php

```diff
@@ -73,3 +73,75 @@ function get_rocket_cdn_cnames( $zone = 'all' ) { // phpcs:ignore WordPress.Nami
 
 	return $hosts;
 }
+
+/**
+ * Check if the current URL is for a live site (not local, not staging)
+ *
+ * @since 3.5
+ * @author Remy Perona
+ *
+ * @return bool True if live, false otherwise.
+ */
+function rocket_is_live_site() {
+	if ( rocket_get_constant( 'WP_ROCKET_DEBUG' ) ) {
```

Comment:
Why does this constant assure us that it's a live site and not local or staging?

Reply:
It doesn't, but we need to see everything when this constant is set, as it's the one we use for debugging.

It's also a constant that is supposed to be used only for a short period of time (debug time), not be kept on forever.

I felt it could fit there in those conditions.

Reply:
If a dev or customer adds that to their staging or local site, then it'll short circuit and give them access.

What if we add an "include" list of domains that can use this constant, such as our staging and dev sites? And we need a way for our CI pipeline to run it too.

Reply:
Or maybe I should ask this:

What is the risk if a dev or customer adds the constant to their staging or local site?

Reply:
We might need to temporarily use this constant on customers website to do debugging, so an allowlist would restrict us too much.

The risk is that they will see the RocketCDN banners and call to action, so they could potentially try to subscribe for those sites if they go through all the process on purpose.

Reply:
Both are true. Let's monitor it and see if we need adjust.